City-of-Helsinki/helsinki-paatokset
# UHF-9346: New minutes PDF field for decisions

# [UHF-9346](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9346)

## What was done
* Add a new minutes PDF field decisions
  * On office holder decisions, this field is used as the main PDF version ("Lataa tulostusversio"-button)
* Add the new field in decision migrations and include a script for updating earlier decisions where the field is missing
* Add update script to cron jobs (100 decisions updated every 10 minutes or so)
* Fix office maker cache invalidation when decisions are reindexed (includes "Luottamushenkilöpäättäjät" now)
* Misc fix for policymaker organisation path margins

## Replicate previous situation
* On test, pull in some test data:
```
drush ap:update decisions {B4B8E27F-9763-4FD0-810C-96ADA6C1C4B7} -v
drush ap:update decisions {ECBF1696-AFFA-42B0-89F4-5E75E9360076} -v
drush ap:update decisions {BA8BE048-BC9B-4B35-A245-AD596889ED1D} -v
drush ap:update decisions {1146484D-75D0-4828-BF9C-9745B7B4B960} -v
``` 
* Open the decision pages:
  - https://helsinki-paatokset.docker.so/fi/asia/hel-2016-012511/b4b8e27f-9763-4fd0-810c-96ada6c1c4b7
  - https://helsinki-paatokset.docker.so/fi/asia/hel-2023-007241/ba8be048-bc9b-4b35-a245-ad596889ed1d
  - https://helsinki-paatokset.docker.so/fi/asia/hel-2023-014372/ecbf1696-affa-42b0-89f4-5e75e9360076
* For each, the "Lataa tulostettava versio" button should lead to the FileURI defined inside the "Decision Record" field's JSON.
* Also, the "Decision history PDF" field should only have the links array, no other data (or it should be empty).

## How to install
* Make sure your instance is up and running on correct branch
  * `git checkout UHF-9346-new-minutes-pdf-field`
  * `make drush-cr drush-cim`
* Run the previous integration commands again

## How to test
* [x] When you edit each node, the "Decision minutes PDF" field should be available and it should have data.
* [x] The "Decision history PDF" field should also have the full JSON object now, or it should be empty / null.
* [x] When you reload the decision pages, the first two city council decisions should have the same the same links, but the other two should have links to the minutes PDF.
* Run this command a few times: `drush ap:ud --logic=minutes -v --limit=100` with different limit values
  * Check some of the recently updated decision nodes: https://helsinki-paatokset.docker.so/fi/admin/content/
  * [x] Each updated node should have data in the minutes PDF field
  * [x] If the node is an office holder decision (organisation type is "Viranhaltija" or "Luottamushenkilö"), the button should point to the minutes PDF, otherwise it should still point to the record PDF.

[UHF-9346]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
